cf_units/__init__.py

```diff
@@ -1514,7 +1514,8 @@ def _op_common(self, other, op_func):
             raise ValueError("Cannot %s a 'no-unit'." % op_label)
 
         if self.is_unknown() or other.is_unknown():
-            result = _Unit(_CATEGORY_UNKNOWN, None)
+            result = _Unit(_CATEGORY_UNKNOWN, _ud.NULL_UNIT,
```

Comment:
This is the crux of the issue. ``_Unit`` is an optimisation that is there when we know the UT handle, but it is highly obfuscated and should be much more explicit. I've addressed this in #131.